packages/material-ui-styles/src/ThemeProvider/ThemeProvider.js

```diff
@@ -5,6 +5,33 @@ import ThemeContext from '../useTheme/ThemeContext';
 import useTheme from '../useTheme';
 import nested from './nested';
 
+export function isPlainObject(item) {
+  return item && typeof item === 'object' && item.constructor === Object;
+}
+
+export function mergeThemes(target, source, options = { clone: true }) {
```

Comment:
Why I went with new method in the end is, because I am adding it on a place where deepmerge was not even used... I can add an options param for this special case to `deepmerge`